Sources/NIO/NonBlockingFileIO.swift

```diff
@@ -189,10 +261,17 @@ public struct NonBlockingFileIO {
             var bytesRead = 0
             while bytesRead < byteCount {
                 let n = try buf.writeWithUnsafeMutableBytes(minimumWritableBytes: byteCount - bytesRead) { ptr in
-                    let res = try fileHandle.withUnsafeFileDescriptor { descriptor in
-                        try Posix.read(descriptor: descriptor,
-                                       pointer: ptr.baseAddress!,
-                                       size: byteCount - bytesRead)
+                    let res = try fileHandle.withUnsafeFileDescriptor { descriptor -> IOResult<ssize_t> in
+                        if let offset = fromOffset {
+                            return try Posix.pread(descriptor: descriptor,
+                                                   pointer: ptr.baseAddress!,
+                                                   size: byteCount - bytesRead,
+                                                   offset: off_t(offset + bytesRead))
```

Comment:
Perhaps, something like `offset.addingReportingOverflow(bytesRead)` would be more correct?

Reply:
Hmm, Int as well as off_t are 64 bit on all platforms we really care about, right? So this should never overflow with reasonable `offset` and if it does I think crashing is fine. On 32 bit this is a bit more nuanced so we could do `off_t(offset) + off_t(bytesRead)`

Reply:
changed to the latter